File: src/platform/graphics/viewport.cpp
#include "sani/platform/graphics/viewport.hpp"

namespace sani {
	namespace graphics {
		
		Viewport::Viewport(const int32 x, const int32 y, const uint32 width, const uint32 height) : x(x),
																									y(y),
																									width(width),
																									height(height) {
		}
		Viewport::Viewport(const Viewport& viewport) : x(viewport.x),
													   y(viewport.y),
													   width(viewport.width),
													   height(viewport.height) {
		}
		Viewport::Viewport() : x(0),
							   y(0),
							   width(0),
							   height(0) {
		}

		Viewport::~Viewport() {
		}
	}
}

File: vs2013/cs/SaNi.Mono/Graphics/Viewport.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SaNi.Mono.Graphics
{
    /// <summary>
    /// Viewport of the device.
    /// </summary>
    public struct Viewport
    {
        #region Fields
        public int x;
        public int y;

        public uint width;
        public uint height;
        #endregion

        public Viewport(int x, int y, uint width, uint height)
        {
            this.x = x;
            this.y = y;

            this.width = width;
            this.height = height;
        }

        public static Viewport Empty()
        {
            return new Viewport(0, 0, 0, 0);
        }
    }
}
